test/test_gc.py

```diff
@@ -17,6 +19,7 @@ def test_gc(self):
     del a,b
     assert(tensors_allocated() == 1) # one for Tensor._rng_counter
 
+  @unittest.skipIf(not CI, "need multiprocessing to work")
```

Comment:
I don't understand why this should fail not in CI. In general, GC failure is caused by another test failing and not cleaning up.

Reply:
I will try to show. (with everything else applied, and not skipping gc tests)
This is local. The problem seems related to the test suite not being run with number of workers >=2, (which you do in CI)
And running only `test/test_gc.py` of course works.

The reason i chose `skipIf(not CI` is that i originally wanted to check if the tests were run with `-n` but that was, afaik, not so nice and easy to do.

Doing some bisecting, i tried to move around(renaming) `test_gc.py` and it seems the problem is in `test/models/test_end2end.py` (of course verify)
Meaning, i rename it to `test/models/test_e1gc.py` so it runs before `test/models/test_efficientnet.py` gc test pass. renaming it to `test/models/test_exgc.py` gc test fails.

If i commenting away:
https://github.com/tinygrad/tinygrad/blob/828c98d5c41e6b3cda97475f3c6910e184f07116/test/models/test_end2end.py#L67-L69
ofcourse fails that test, but `test_gc.py` passes

without touching `test/models/test_end2end.py`
failing (its only the gc tests that shows as failed)
```
$ python3 -m pytest test/     
========================================================== test session starts ===========================================================
platform darwin -- Python 3.8.19, pytest-8.2.2, pluggy-1.5.0
rootdir: /Users/willygorillatg/tinygrad
plugins: hypothesis-6.103.1, xdist-3.6.1
collected 1782 items                           
```
.....
```
======================================================== short test summary info =========================================================
FAILED test/test_gc.py::TestGC::test_gc - ReferenceError: weakly-referenced object no longer exists
FAILED test/test_gc.py::TestGC::test_gc_complex - ReferenceError: weakly-referenced object no longer exists
=========================== 2 failed, 1567 passed, 177 skipped, 36 xfailed, 192 warnings in 141.32s (0:02:21) ======
```

ok
```
$ python3 -m pytest -n=2 test/
========================================================== test session starts ===========================================================
platform darwin -- Python 3.8.19, pytest-8.2.2, pluggy-1.5.0
rootdir: /Users/willygorillatg/tinygrad
plugins: hypothesis-6.103.1, xdist-3.6.1
2 workers [1782 items]  
```
.....
```
================================= 1569 passed, 177 skipped, 36 xfailed, 222 warnings in 95.87s (0:01:35) ===
```

Reply:
yes, without -n= it will fail. But `not CI` doesn't look like the right fix.

Reply:
the easiest fix, is to add `-n=auto` to the README.md 
```
python3 -m pytest test/                 # whole test suite
```
to
```
python3 -m pytest -n=auto test/                 # whole test suite
```